src/utilities/convex_hull.zig

```diff
@@ -202,8 +250,52 @@ pub const ConvexHull = struct {
         return hull;
     }
 
-    /// Auxiliary function to add a new `point` to a given `hull` of the convex hull. The function uses
-    /// the given `turn` to correctly add the new point.
+    /// Helper function to get all points sorted as an `ArrayList(DiscretePoint)`. The function
+    /// merges the lower and upper hulls, which are already sorted, in O(N) time. If the
+    /// `ConvexHull` is empty, an empty ArrayList is returned.
+    pub fn getAllPointsSorted(self: *ConvexHull) !ArrayList(DiscretePoint) {
+        // Initialize the result list with enough capacity for both hulls
+        var all_points = try ArrayList(DiscretePoint).initCapacity(
+            self.allocator,
+            self.lower_hull.items.len + self.upper_hull.items.len - 2,
+        );
+
+        var lower_idx: usize = 0;
+        var upper_idx: usize = 1;
+
+        // Merge lower_hull and upper_hull into all_points
+        while (lower_idx < self.lower_hull.items.len and upper_idx < self.upper_hull.items.len - 1) {
+            const lower_point = self.lower_hull.items[lower_idx];
+            const upper_point = self.upper_hull.items[upper_idx];
+
+            if (lower_point.time <= upper_point.time) {
+                // Add the point from the lower hull
```

Comment:
Missing punctuation at the end of the comment.

Reply:
fixed.